modules/dnn/src/layers/prior_box_layer.cpp

```diff
@@ -314,6 +314,10 @@ class PriorBoxLayerImpl CV_FINAL : public PriorBoxLayer
         std::vector<UMat> inputs;
         std::vector<UMat> outputs;
 
+        bool use_half = (inps.depth() == CV_16S);
```

Comment:
PriorBox layer does not use input data. It generates a fixed set of bounding boxes. So I think we need to keep it's output in single precision floats because we can face significant accuracy loss.

Reply:
thanks for the review,  I keep the float precision for prior_box layer.